### PR TITLE
feat/add-custom-agents-support

### PR DESCRIPTION
**Pull request by Sadzurami**
Add ability to pass custom agents. Useful if required to use one agent for all steam requests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,7 +254,9 @@ event is emitted.
 		If omitted, defaults to a `WebSocketCMTransport` instance for `SteamClient` platform types, and a 
 		`WebApiTransport` instance for all other platform types. In all likelihood, you don't need to use this.
     - `httpProxy` - A string containing a URI for an HTTP proxy. For example, `http://user:pass@1.2.3.4:80`
-    - `socksProxy` A string containing a URI for a SOCKS proxy. For example, `socks5://user:pass@1.2.3.4:1080`
+    - `socksProxy` - A string containing a URI for a SOCKS proxy. For example, `socks5://user:pass@1.2.3.4:1080`
+    - `agent` - An `https.Agent` instance to use for requests. If omitted, a new `https.Agent` will be created
+      internally.
```
Comment (DoctorMcKay) [inline README.md:259]: The docs should mention that you can only use one of `httpProxy`, `socksProxy`, or `agent` at the same time.

File: src/LoginApprover.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export default class LoginApprover {
 	private _handler: AuthenticationClient;
 
 	constructor(accessToken: string, sharedSecret: string|Buffer, options?: ConstructorOptions) {
-		let agent:HTTPS.Agent = new HTTPS.Agent({keepAlive: true});
+		let agent:HTTPS.Agent = options.agent || new HTTPS.Agent({keepAlive: true});
 		if (options.httpProxy && options.socksProxy) {
 			throw new Error('Cannot specify both httpProxy and socksProxy at the same time');
```
Comment (DoctorMcKay) [inline src/LoginApprover.ts:25]: I'd like to see this check updated to enforce that only one of `httpProxy`, `socksProxy`, or `agent` is used.

File: src/LoginSession.ts [hunk 3]
```diff
@@ -104,7 +104,7 @@ export default class LoginSession extends TypedEmitter<LoginSessionEvents> {
 
 		options = options || {};
 
-		let agent:HTTPS.Agent = new HTTPS.Agent({keepAlive: true});
+		let agent:HTTPS.Agent = options.agent || new HTTPS.Agent({keepAlive: true});
 		if (options.httpProxy && options.socksProxy) {
 			throw new Error('Cannot specify both httpProxy and socksProxy at the same time');
```
Comment (DoctorMcKay) [inline src/LoginSession.ts:109]: Same update to this check here.

**Review by DoctorMcKay — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Sadzurami) [reply README.md:259]: > The docs should mention that you can only use one of `httpProxy`, `socksProxy`, or `agent` at the same time.

Done

File: src/LoginApprover.ts (hunk 2 above)
Comment (Sadzurami) [reply src/LoginApprover.ts]: > I'd like to see this check updated to enforce that only one of `httpProxy`, `socksProxy`, or `agent` is used.

Done

File: src/LoginSession.ts (hunk 3 above)
Comment (Sadzurami) [reply src/LoginSession.ts]: > Same update to this check here.

Done

**Pull request merged**